_adr/81/index.adoc

```diff
@@ -19,10 +19,267 @@ NOTE: https://datatracker.ietf.org/doc/html/rfc2119[RFC-2119] keywords are used
+For example a shard on AWS might support a different set of profiles to a shard on Azure, or the same profiles might have different sizes on those clouds.
+One of the purposes of introducing the profile/size concept is to provide flexibility in the kinds of service offered in the future.
+
+* The availability of a profile might be limited to a particular cloud, region or combination of both.
+
+* It is a private concern of the shard exactly how each size in a profile is provisioned.
+In principle it could use vertical and/or horizontal scaling and/or density > 1, in addition to applying quotas to limit capacity to the advertised limits.
+The fleet manager doesn't know.
+
+=== UI/Fleet manager interface
+
+This ADR does not mandate any particular presentation for how the scale options are presented in the UI.
+In particular while there is only one profile defined, there is no requirement to display profiles in the UI at all.
+
+The fleet manager SHALL be the source of truth for the profiles and sizes available, and how those depend on cloud and region. 
+The fleet manager SHALL provide an endpoint which the UI MUST use to determine the available profiles and sizes.
+The UI MUST NOT make assumptions about linearity of scaling, instead it uses only the profiles and sizes provided by the endpoint.
+
+==== Profile discovery
+
+Reusing the "Standard" profile defined above, let's illustrate how the Fleet Manager could communicate the available kinds of cluster to a UI.
+
+.Sequence diagram of UI/Fleet manager interaction
+image::adr-81-ui-fleet_manager.png[UI/Fleet manager interaction]
+
+After the UI has discovered the available cloud providers and regions it would make another GET request `/api/kafkas_mgmt/v1/profile/{cloudId}/{cloudRegion}` to discover the profiles available in this cloud and region.
+An example response JSON is shown below.
+
+[source,json]
+.Example profile available to a UI from the Fleet Manager's `GET /api/kafkas_mgmt/v1/profiles/{cloudId}/{cloudRegion}` endpoint
+----
+"profiles": [ <1>
+    { "id": "standard", <2>
+      "parameters": { <3>
+        "ingressThroughputPerSec" : { "name": "Ingress", "unit": "MiB/s" },
+        "egressThroughputPerSec": { "name": "Egress", "unit": "MiB/s" },
+        "maxPartitions": { "name": "Partitions" },
+        "totalMaxConnections": { "name": "Connections" },
+        "maxConnectionAttemptsPerSec": { "name": "Connection Rate", "unit": "/s" },
+        "maxMessageSize": { "name": "Maximum message size", "unit": "MiB" },
+        "cost": { "name": "Number of entitlements required to create an instance" }
+      },
+      "sizes": [ <4>
+        { "id": "s", <5>
+          "parameterValues": { <7>
+            "ingressThroughputPerSec": "30", "egressThroughputPerSec": "30", 
+            "maxPartitions": "1000", "totalMaxConnections": "500",
+            "maxConnectionAttemptsPerSec": "100", "maxMessageSize": "1",
+            "cost": 1,
```

Comment:
The only concern about this is that `cost` is an internal detail of the fleet manager (Quota cost). I'm not sure if we should expose it here. What do you think @ziccardi ?

Reply:
Ah, well if it's internal then no need to expose it in the contract between the UI and the fleet manager. That doesn't prevent us passing it into the fleet manager from app interface though. But I guess I'd still like to know whether/how a notion of cost gets into the UI. @pmuir probably knows.

Reply:
We have to put the size somewhere: we can enforce some naming on the sizes id ('x1', 'x2', etc) or add a `cost` attribute: IMO we will expose the detail either way.
Having to choose one, I'd prefer to have the `cost` attribute.

Reply:
I think that's fine for the app-interface/fleet manager contract, and omit that parameter in the contract between fleet manager and its clients.